### DISCUSSION TITLE
Had break in need to download footage

### DISCUSSION THREAD

**Discussion by YBonline** [blakeblackshear/frigate#6665] [General]
Hi,

I had a breakin 3 hours ago and Frigate got it on video and I need to download the footage for the police. I can see portions of it on the event viewer and download them, but they covered there faces with shits and the object recognition no longer recognized them as people. I can see that it was recorded in the recordings viewer, but there is apparently no option to download in there. Is there an easy way to download the entire hour? I found instructions to use the API with a start and end time, and tried this URL:
http://cameras/api/gp_bldg_08_front/start/1685578739/end/1685586129/clip.mp4

I found the 10 second clip files in the folder 2023-06-01/00/gp_bldg_08_front but all I get when I try to download the clip is: Could not create clip from recordings for gp_bldg_08_front.

The time I need is between 8:15 and 9:15pm EST

Also any chance to add a download button on each hour for the recordings section so its really easy and simple to download the video in an emergency?

**Comment by NickM-27** [comment]
You just need to use the api for now, it is in Unix Timestamps. The next version will have a recordings exporter.

**Comment by YBonline** [comment]
Apparently when I try to download the file the cache directory is filling up :-/

**Reply by blakeblackshear** [reply]
Use the m3u8 endpoint with ffmpeg.
https://github.com/blakeblackshear/frigate/issues/2369#issuecomment-1398023365

**Comment by der--flo** [comment]
you cold also try to combine the hourly recordings. create a file, e.g. recording.txt:

```
file 'http://frigate-ip:5000/vod/2023-06/01/8/<camera>/master.m3u8'
file 'http://frigate-ip:5000/vod/2023-06/01/9/<camera>/master.m3u8'
```
and run ffmpeg to combine the segments from the m3u8's into one .mp4:

`ffmpeg -y -safe 0 -protocol_whitelist "file,http,https,tcp,tls,hls,concat" -f concat -i recording.txt -c copy ./output.mp4`
